packages/cli/src/util/projects/connect-git-provider.ts

```diff
@@ -0,0 +1,101 @@
+import Client from '../client';
+import { stringify } from 'qs';
+import { Team } from '../../types';
+import chalk from 'chalk';
+
+export async function disconnectGitProvider(
+  client: Client,
+  team: Team | null,
+  projectId: string
+) {
+  const fetchUrl = `/v4/projects/${projectId}/link?${stringify({
+    teamId: team?.id,
+  })}`;
+  return client.fetch(fetchUrl, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+}
+
+export async function connectGitProvider(
+  client: Client,
+  team: Team | null,
+  projectId: string,
+  type: string,
+  repo: string
+) {
+  const fetchUrl = `/v4/projects/${projectId}/link?${stringify({
+    teamId: team?.id,
+  })}`;
+  return client
+    .fetch(fetchUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        type,
+        repo,
+      }),
+    })
+    .catch(err => {
+      if (err.message.includes('install the GitHub integration')) {
```

Comment:
Does the error not contain a `code` or more reliable property that we can check to infer these states?

Reply:
Both of these return 400. The message is the only reliable thing I can think of to differentiate them. But I'm not sure.

Reply:
For posterity: Nate and I opened & merged a PR to vercel/api to address this. https://github.com/vercel/api/pull/12991